### PR TITLE
improve(amplitude): instrument bridge buttons

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR is aimed at instrumenting the bridge button clicks with a custom event. In order to get the environment configured to add this event, the Amplitude tracking had to be converted to a provider/context model.

### CODE REVIEW

File: src/app/_components/bridge-now-link.tsx [hunk 1]
```diff
@@ -15,14 +17,24 @@ export function BridgeNowLink(props: Props) {
...
-function _BridgeNowLink({ className, ...props }: Props) {
+function _BridgeNowLink({ className, section, ...props }: Props) {
   const searchParams = useSearchParams();
 
   const refParams = searchParams.get("ref") || searchParams.get("referrer");
   const bridgeNowLink = `${bridgeAppBaseUrl}/bridge${refParams ? `?ref=${refParams}` : ""}`;
 
+  const pathname = usePathname();
+  const actionCallback = useCallback(() => {
+    console.log("HI(");
```
Comment (dohaki) [inline src/app/_components/bridge-now-link.tsx:28]: Can be removed?

File: src/app/_components/bridge-now-link.tsx [hunk 2]
```diff
@@ -15,14 +17,24 @@ export function BridgeNowLink(props: Props) {
...
 
-function _BridgeNowLink({ className, ...props }: Props) {
+function _BridgeNowLink({ className, section, ...props }: Props) {
   const searchParams = useSearchParams();
 
   const refParams = searchParams.get("ref") || searchParams.get("referrer");
   const bridgeNowLink = `${bridgeAppBaseUrl}/bridge${refParams ? `?ref=${refParams}` : ""}`;
 
+  const pathname = usePathname();
+  const actionCallback = useCallback(() => {
```
Comment (dohaki) [inline src/app/_components/bridge-now-link.tsx:27]: Is this memoization really needed?

**Review by dohaki — APPROVED** [review]
LGTM 👍 Left two small nits

File: src/app/_components/bridge-now-link.tsx (hunk 2 above)
Comment (james-a-morris) [reply src/app/_components/bridge-now-link.tsx]: Fair - I've removed it in place of a lambda function.

**Pull request merged**